haskell-lsp-types/src/Language/Haskell/LSP/Types/Message.hs

```diff
@@ -380,30 +381,58 @@ deriveJSON lspOptions{ fieldLabelModifier = customModifier } ''ResponseError
 
 -- ---------------------------------------------------------------------
 
--- | Either result or error must be Just.
+{-
+  https://microsoft.github.io/language-server-protocol/specification#responseMessage
+
+  interface ResponseMessage extends Message {
+    /**
+    * The request id.
+    */
+    id: number | string | null;
+
+    /**
+    * The result of a request. This member is REQUIRED on success.
+    * This member MUST NOT exist if there was an error invoking the method.
+    */
+    result?: string | number | boolean | object | null;
+
+    /**
+    * The error object in case a request fails.
+    */
+    error?: ResponseError;
+  }
+-}
 data ResponseMessage a =
   ResponseMessage
-    { _jsonrpc :: Text
-    , _id      :: LspIdRsp
-    , _result  :: Maybe a
-    , _error   :: Maybe ResponseError
+    { _rmjsonrpc :: Text
+    , _rmid      :: LspIdRsp
+    , _rmresult  :: Either ResponseError a
     } deriving (Read,Show,Eq)
```

Comment:
We have been naming the fields to correspond with the spec, with an underscore to allow idiomatic lens generation.  The `OverloadedRecordFields` extension allows us to do this.

So I think it would be better to stick with the current usage, i.e. get rid of the `rm` bit in the field names.

Reply:
Thanks! Got it. I'll remove them. 
I've added `rm` prefixes because compiler errored on `toJson` :
```haskell
instance ToJSON a => ToJSON (ResponseMessage a) where
  toJSON r = object
    [ "jsonrpc" .= _rmjsonrpc r
    , "id" .= _rmid r
    , "result" .= rightToMaybe (_rmresult r)
    , "error" .= leftToMaybe (_rmresult r)
    ]
``` 
with functions (record fields) being ambigous. I'll find another way.

Reply:
Done.